src/struct/commands/arguments/Argument.ts

```diff
@@ -311,8 +311,9 @@ export class Argument {
 					const sentTimeout = await message.channel.send(timeoutText);
 					if (message.util) message.util.addMessage(sentTimeout);
 				}
-
-				return Flag.cancel();
+				const time = promptOptions?.time ? this.command.argumentDefaults.prompt?.time : 30000;
+				if (!time) return Flag.cancel();
+				return Flag.timeout(time);
```

Comment:
`promptOptions` will always be non-nullish and `promptOptions.time` is already set to its default earlier. 
```suggestion
				if (!promptOptions.time) return Flag.cancel();
				return Flag.timeout(promptOptions.time);
```